Automattic/WP-Job-Manager
# Rename licence -> license throughout codebase

Built off of #2552 

While working on #2552, I realized we have a very brittle setup with the usage of `licence` vs (US) `license`. This finally standardizes it to use `license` and migrates the licenses appropriately. Our strings should previously all been updated, but this switches over the code/options.

### Changes proposed in this Pull Request

* Migrate `licence` to `license` in code and options.

### Testing instructions

* On `trunk`, activate licenses for products.
* Switch to this branch (note: needs an asset rebuild) and ensure licenses aren't lost. Try activating, deactivating, etc and see how it goes.


### Deprecated Code

* `WP_Job_Manager_Helper::has_licenced_products` -> `WP_Job_Manager_Helper::has_licensed_products`
* `WP_Job_Manager_Helper::get_plugin_licence` -> `WP_Job_Manager_Helper::get_plugin_license`
* `WP_Job_Manager_Helper::licence_output` -> `WP_Job_Manager_Helper::licence_output`
* `WP_Job_Manager_Helper::activate_licence` -> `WP_Job_Manager_Helper::activate_license`
* `WP_Job_Manager_Helper::deactivate_licence` -> `WP_Job_Manager_Helper::deactivate_license`
